code-injection/row_007_AzisabaNetwork_SpicyAzisaBan-API_users.ts

```
@@ -0,0 +1,260 @@
+  validate2FAToken,
+  generateSecureRandomString,
+  w, getSessionKey, deleteSession, getSession, getUser,
+} from '../util/util'
+import { createNew } from '../util/totp'
+export const router = express.Router()
+
+router.post('/check_email', w(async (req, res) => {
+  if (!req.body || typeof req.body !== 'object') return res.status(400).send({ error: 'invalid_params' })
+  const user_id = parseInt(req.body.user_id)
+  const email = String(req.body.email)
+  if (!user_id || user_id !== user_id || user_id < 0 || !email || !email.includes('@')) return res.send({ valid: false })
```

Biome (2.1.2)

[error] Comparing to itself is potentially pointless.



(lint/suspicious/noSelfCompare)